workflow/controller/exec_control.go

```diff
@@ -82,10 +82,23 @@ func (woc *wfOperationCtx) handleExecutionControlError(nodeID string, wfNodesLoc
 	node := woc.wf.Status.Nodes[nodeID]
 	woc.markNodePhase(node.Name, wfv1.NodeFailed, errorMsg)
 
+	children := []wfv1.NodeStatus{}
```

Comment:
This looks like a useful utility function,:

- Is there already a similar function?
- If not, extract it, e.g. `woc.wf.Status.Nodes.GetTransientChildren(nodeID)`

Extracting it to a function will:

- Allow reuse.
- Write test.